### ISSUE TITLE
When players_per_group = None

### ISSUE THREAD

**Issue by ScottClaessens** [chapkovski/custom-waiting-page-for-mturk#6]
Hello,

The participant counts (number arrived, number waiting for, total number) do not seem to work on this Custom Waiting Page if there is one large group in the session (i.e. players_per_group = None). Is there a workaround for this?

Thanks,
Scott

**Comment by ngruyer** [comment]
Hi Scott.

Can you explain your need? Would you like to have a wait page that only goes to the next step if all the participants of the session have arrived, but that in between allows participants to leave the experiment? (meaning that inside the experiment you will have a different number of active participants, depending on the number of dropouts)
Or something else?

For now, the logic of the Custom Mturk Waiting Page page is based on the idea that the experiment needs a fixed number of active participants in each group, so it cannot work with "players_per_group = None". 

Nicolas

**Comment by ScottClaessens** [comment]
Hi Nicolas,

I'm sorry for my very late reply to this. I've managed to find a quick-and-dirty solution myself in the meantime, but I'll still explain my need anyway.

I have a survey-like app that all individuals in the session complete individually. Then, they must wait for everyone else in the session to finish the survey before continuing the study on their own (the responses of others are randomly 'matched' to determine payoffs). The main thing I wanted to lift from this CustomWaitPage was the count of 'number of participants arrived, out of [session size]' in the wait page, so that participants know how many people they have to wait for. My quick-and-dirty solution was to simply change Constants.players_per_group to len(self.subsession.get_players()) in the wait page template and consumers.py. It seems to work, for my needs. This particular study is online, but not done through MTurk.

Thanks again for creating this fantastic wait page, I've used it in other MTurk studies and it worked fantastically.

Best,
Scott
